### ISSUE TITLE
issue when saving a null value

### ISSUE THREAD

**Issue by jamesdixon** [seegno/bookshelf-json-columns#31]
Hello again!

I ran into an issue with the 1.2.1 release. I specifically noticed it when I was trying to save a payload that contained a property that had a value of `null`. For example, if I was to call `model.save(data)` where `data` contains something like:

```js
{
  preferences: null
}
```

Before saving, I have a method that validates the model, which is tapping into the `save` hook provided by Bookshelf. When inspecting the model that's passed into that function, I can see that `preferences` is now a string:

```js
{
  preferences: "null"
}
```

I'm assuming this is because it's stringified beforehand, but ultimately, this is causing problems on my end.

If I remove `preferences` as a JSON column, it works as expected. Reverting to v1.1.1 also resolves the issue.

Thanks!

**Comment by jamesdixon** [comment]
A couple of additional data points:

1. This only occurs when `{ patch: true}`
2. If you run the following test and then look at the value of the `foo` column in the test table, you'll see that it contains a `"null"` value.

```js
    it.only('null or undefined values should not be stringified', async () => {
      const model = await Model.forge().save();

      await model.save({ foo: null }, { patch: true });

      should(model.get('foo')).be.null();
    });
```

Regarding the test, this test actually passes. It appears that either Bookshelf, Knex or the PG driver coerces the `null` string into an actual `null` value.

**Comment by jamesdixon** [comment]
I believe I have a fix for this: https://github.com/scoutforpets/bookshelf-json-columns/commit/d91e9f5866697cfb784392564f5529753cc5ee57

Let me know what you think. 

The issue now is I'm unsure how to test since something is coercing the `null` string mentioned above into an actual `null` value. Any ideas?

**Comment by ricardogama** [comment]
Sorry for the delay @jamesdixon, already have a fix for this. Thanks again for contributing!

**Comment by jamesdixon** [comment]
Awesome. Thanks!

**Comment by ricardogama** [comment]
@jamesdixon Released as [1.2.2](https://github.com/seegno/bookshelf-json-columns/releases/tag/1.2.2), cheers!
